GRDB/Record/FetchableRecord+Decodable.swift

```diff
@@ -79,7 +81,30 @@ private struct RowKeyedDecodingContainer<Key: CodingKey>: KeyedDecodingContainer
             } else if dbValue.isNull {
                 return nil
             } else {
-                return try T(from: RowDecoder(row: row, codingPath: codingPath + [key]))
+                do {
+                    // This decoding will fail for types that need a keyed container,
+                    // because we're decoding a database value here (string, int, double, data, null)
+                    // if we find a nested Decodable type then pass the string to JSON decoder
+                    let value = try T(from: RowDecoder(row: row, codingPath: codingPath + [key]))
+                    if let _ = value as? Codable {
+                        // Support for nested ( Codable )
+                        return try JSONDecoder().decode(type.self, from: row.dataNoCopy(named: key.stringValue)!)
```

Comment:
I don't understand the reason for `if let _ = value as? Codable`. The comment `Support for nested ( Codable )` doesn't help me understand.

- we are decoding *twice* here, which looks like a big waste of time.

- why do we check for `Codable`, since we are only decoding? Isn't `Decodable` enough?

- how on Earth this condition could be false, since we are decoding a `Decodable` type?

All this code is pretty obscure to me. What do I miss?

Reply:
I am not sure this is the best but the way the current code is basically considering the Encodable compliance a such 4 tests hit this code the following way :
if let _ = value as? Codable {
       //testOptionalNestedStruct
} else {
       //testDecodableRawRepresentableProperty
       //testNonTrivialSingleValueDecodableProperty
       //testTrivialSingleValueDecodableProperty
}

Reply:
Thanks for the details. Please provide an answer to other questions as well.

Reply:
I want you to understand, @valexa, that the `if let _ = value as? Codable` test **has to go**.

That's because it would have GRDB behave differently if I start decoding this type:

```swift
struct MyStruct: Decodable { ... }
```

And then I decide that MyStruct should be Encodable as well:

```swift
struct MyStruct: Codable { ... }
```

In the first version of MyStruct, `if let _ = value as? Codable` doesn't pass. In the second version, `if let _ = value as? Codable` passes. Decoding will change. Just because I added a conformance to MyStruct, *totally unrelated to decoding*.

**This is an important bug that has to be fixed.**

Reply:
I've downloaded your code, and run the tests. I see the problem you are facing. Please hold on until I find the proper way out.

Reply:
Done